ocaml/xapi/xapi_network.ml

```diff
@@ -329,3 +329,40 @@ let with_networks_attached_for_vm ~__context ?host ~vm f =
         error "Caught %s while detaching networks" (string_of_exn e)
     end;
     raise e
+
+(* Note that in the revision history is a version of this function
+ * with logic for additional purposes, which may be useful in future. *)
+let assert_can_add_purpose ~__context ~network ~current newval =
+  let sop (*string-of-porpoise*) = Record_util.network_purpose_to_string in
+  let reject str =
+    raise Api_errors.(Server_error (network_incompatible_purposes, [ (sop newval); str ]))
```

Comment:
We could have 
```
raise Api_errors.(Server_error (network_incompatible_purposes, [ (sop newval); (sop conflicting) ]))
```
because both must be purposes - that is how this error is defined.